### DISCUSSION TITLE
Extract appearance state of widgets

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#1700] [Looking for help]
Can you let me know how appearance state of widgets can be extracted?

**Comment by JorjMcKie** [comment]
> Can you let me know how appearance state of widgets can be extracted?

What do you mean? Look at [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/fields/list-fields.py) demo to see what be extracted for a field.

**Comment by meghanaviyyapu** [comment]
![capture1](https://user-images.githubusercontent.com/36301306/166103958-a6f3e649-eb0a-44f5-a84d-6eefa3fd2ee9.JPG)
I want to extract the values of appearance of widgets(/AP).
In PyMuPDF I have seen methods to extract border attributes, field attributes and text related attributes for widgets. Can you let me know how to extract appearance related attributes?

**Comment by JorjMcKie** [comment]
The script I mentioned is the maximum of what be done with pymupdf.

Holen Sie sich Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by meghanaviyyapu** [reply]
Can you let me know if a new PDF with same widgets can be created after extracting widget information from an existing PDF without changing the appearance of widgets in the new PDF?

**Reply by JorjMcKie** [reply]
No, this will not work in all those cases where the old widget was not created with PyMuPDF.

**Reply by meghanaviyyapu** [reply]
Ok Thanks. 
Do you have any idea how lists can be extracted from a PDF?

**Reply by JorjMcKie** [reply]
What do you mean lists?

**Reply by meghanaviyyapu** [reply]
![Capture](https://user-images.githubusercontent.com/36301306/166444403-64556cbd-f081-4b9e-b417-10d276d80683.JPG)
Ordered and Unordered lists like these.

**Reply by JorjMcKie** [reply]
This is the same topic as one of your earlier questions about tables, headers and footers:
> No, this is just regular text. Just must develop your own code to decipher it.
